mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -217,9 +219,13 @@ class ConvConverter final : public OpConversionPattern<tosa::Conv2DOp> {
     FailureOr<rock::Conv2DOp> rockConv = makeRockConv2D(
         rw, op, input, inputLayout, filter, filterLayout, output, outputLayout,
         op.getPad(), op.getStride(), op.getDilation());
+
     if (failed(rockConv))
       return failure();
 
+    // disable layout changes since will be transforms
```

Comment:
Hang on,
So now, it doesn't do high-level layout optimization and just fuse transposes as transforms?
For example, MIGraphX always has NCHW layout and tosa only uses NHWC, so the conversion always includes transforms on both input/output.
This changes memory access  pattern of all LD/ST and probably kill the performance.
Even though somehow the transforms can be well optimized at the end, we don't have a scheme to tune the kernel with the  fused transforms at the moment.

I'm concerning this can significantly degrade the performance of current MIGraphX integration and would suggest to hold this part of change until we can evaluate the performance.

Reply:
@krzysz00 believes this should work out as before since it untransforms to determine actual layouts. Is there a way to generate the perf from a PR?

Reply:
Ok, so, all this does (if we're talking about the magic attribute) is stop conv-to-gemm from marking the indices as `Unfold`, which is a funky optimization that would break if you change which dimensions are contiguous in memory that'll be going away soon.

_However_, on to your broader point, this does have the possibility to kill perf, and @sjw36, you may want to write up a "pull transposes into the convolution layout" pass again. The reason for that is that, while untransform() will get you to your underlying out, conv-to-gemm uses the layout specified on the convolution (and not any underlying layout, which can't really be interpreted) to determine how to flatten out the dimensions.

For example, if I tell `rock.conv2d` that I have a KCYX filter, then conv-to-gemm will generate `gemmK <- Merge(C, Y, X)`. *However,*, if I say I have a `KYXC` tensor, we will instead have the transform `gemmK <- Merge(Y, X, C)`, on the assumption that you're generally calling `rock.conv2d` with a tensor whose layout is basically like what you passed as an argument to `rock.conv2d`. Therefore, having the transpose just hanging out in the transform stack, while not fatal to correctness, could be bad for perf

Reply:
Added this folding back in.